src/routes/api/data.ts

```diff
@@ -214,21 +214,11 @@ router.get("/getTVProvidersByZipCode", async (req: Request, res: Response) => {
 
     // me when i over complicate things
     if (!zip || !lang || !country) {
-        const query: string[] = [];
-    
-        if (!zip) query.push("zipcode");
-        if (!lang) query.push("lang");
-        if (!country) query.push("country");
-    
-        let missingMessage: string;
-
-        if (query.length === 3) {
-            missingMessage = `${query[0]}, ${query[1]}, and ${query[2]}`;
-        } else if (query.length === 2) {
-            missingMessage = `${query[0]} and ${query[1]}`;
-        } else {
-            missingMessage = query[0];
-        }
+        const missingParams = ["zipcode", "lang", "country"].filter((param, index) => 
+            ![zip, lang, country][index],
+        );
+
+        const missingMessage = missingParams.join(", ").replace(/, ([^,]*)$/, " and $1");
```

Comment:
In a "general repo documentation" we should probably explain what this does

Reply:
it just checks whats parts of the body are mising

Reply:
yeah but the missingParams replace stuff - well ig its regex